backends/core/workers.py

```diff
@@ -851,14 +851,28 @@ def _send_batch_hits(self, batch_payload, user_agent='CRMint / 0.1'):
     Raises: MeasurementProtocolException if the HTTP request fails.
     """
     headers = {'user-agent': user_agent}
-    req = requests.post('https://www.google-analytics.com/batch',
-                        headers=headers,
-                        data=batch_payload)
+    if self._debug:
+      for payload in batch_payload.split('\n'):
+        response = requests.post('https://www.google-analytics.com/debug/collect',
+                                 headers=headers,
+                                 data=payload)
+        result = json.loads(response.text)
+        if (not result['hitParsingResult'] or
```

Comment:
Could also be written using python built-in functions, as you wish: 
```
if any([not result['hitParsingResult'],
            not result['hitParsingResult'][0]['valid']])
```

Reply:
@dulacp thanks for suggesting `any()` — usually it really helps to make the code more intuitive and more readable.

Yet, I don't think it works well in this specific case: both conditions would be evaluated before calling `any()`. What we actually want is the interpreter to completely skip evaluation of the second condition when the first one evaluates to `True`. Otherwise `IndexError: list index out of range` exception will be thrown when `result['hitParsingResult']` array is empty — that's what we catch with the first condition.